sphinx_gallery/directives.py

```diff
@@ -77,3 +82,200 @@ def run(self):
+            srcset[float(mult)] = spl[0]
+        else:
+            raise ExtensionError('srcset argument "{entry}" is invalid.')
+    return srcset
+
+
+def visit_imgsg_html(self, node):
+
+    if node['srcset'] is None:
+        self.visit_image(node)
+        return
+
+    imagedir, srcset = _copy_images(self, node)
+
+    # /doc/examples/subd/plot_1.rst
+    docsource = self.document['source']
+    # /doc/
+    # make sure to add the trailing slash:
+    srctop = os.path.join(self.builder.srcdir, '')
+    # examples/subd/plot_1.rst
+    relsource = os.path.relpath(docsource, srctop)
+    # /doc/build/html
+    desttop = os.path.join(self.builder.outdir, '')
+    # /doc/build/html/examples/subd
+    dest = os.path.join(desttop, relsource)
+
+    # ../../_images/
+    imagerel = os.path.relpath(imagedir, os.path.dirname(dest))
+    imagerel = os.path.join(imagerel, '')
+    if '\\' in imagerel:
+        imagerel = imagerel.replace('\\', '/')
+    # make srcset str.  Need to change all the prefixes!
+    srcsetst = ''
+    for mult in srcset:
+        nm = os.path.basename(srcset[mult][1:])
+        # ../../_images/plot_1_2_0x.png
+        relpath = imagerel+nm
+        srcsetst += f'{relpath}'
+        if mult == 0:
+            srcsetst += ', '
+        else:
+            srcsetst += f' {mult:1.1f}x, '
+    # trim trailing comma and space...
+    srcsetst = srcsetst[:-2]
+
+    # make uri also be relative...
+    nm = os.path.basename(node['uri'][1:])
+    uri = imagerel + nm
+
+    alt = node['alt']
+    if node['class'] is not None:
+        classst = node['class'][0]
+        classst = f'class = "{classst}"'
+    else:
+        classst = ''
+
+    html_block = (f'<img src="{uri}" srcset="{srcsetst}" alt="{alt}"' +
+                  f' {classst}/>')
+    self.body.append(html_block)
+
+
+def visit_imgsg_latex(self, node):
+
+    if node['srcset'] is not None:
+        imagedir, srcset = _copy_images(self, node)
+        maxmult = -1
+        # choose the highest res version for latex:
+        for key in srcset.keys():
+            maxmult = max(maxmult, key)
+        node['uri'] = str(PurePosixPath(srcset[maxmult]).name)
+
+    self.visit_image(node)
+
+
+def _copy_images(self, node):
+    srcset = _parse_srcset(node['srcset'])
+
+    # where the sources are.  i.e. myproj/source
+    srctop = self.builder.srcdir
+
+    # copy image from source to imagedir.  This is
+    # *probably* supposed to be done by a builder but...
+    # ie myproj/build/html/_images
+    imagedir = os.path.join(self.builder.imagedir, '')
+    imagedir = PurePosixPath(self.builder.outdir, imagedir)
+
+    os.makedirs(imagedir, exist_ok=True)
+
+    # copy all the sources to the imagedir:
+    for mult in srcset:
+        abspath = PurePosixPath(srctop, srcset[mult][1:])
+        shutil.copyfile(abspath, imagedir / abspath.name)
+
+    return imagedir, srcset
+
+
+def depart_imgsg_html(self, node):
+    pass
+
+
+def visit_sg_other(self, node):
```

Comment:
Would anyone be able to explain this function? It does not seem to be used anywhere?
Saw while doing some maintenance dealing with our new linting. Thanks.
cc @jklymak @larsoner

Reply:
This isn't being used by matplotlib, which is the only context where I use sphinx-gallery.  I imagine it was left over from hacking and didn't get cleaned up (so good lint material).

Reply:
Thanks, I'm not so familiar with Sphinx app stuff so wasn't sure if I missed something.